webapp/app/forms/steps/lotse/personal_data.py

```diff
@@ -173,6 +173,107 @@ def render(self, **kwargs):
+            render_kw={'data_label': _l('form.lotse.field_person_idnr.data_label')})
+        person_a_dob = LegacySteuerlotseDateField(
+            label=_l('form.lotse.field_person_dob'),
+            render_kw={'data_label': _l('form.lotse.field_person_dob.data_label')},
+            validators=[InputRequired(message=_l('form.lotse.validation-dob-missing')), ValidDateOfBirth()],
+            prevent_validation_error=True)
+        person_a_first_name = SteuerlotseNameStringField(
+            label=_l('form.lotse.field_person_first_name'),
+            render_kw={'data_label': _l('form.lotse.field_person_first_name.data_label'),
+                       'max_characters': 25},
+            validators=[InputRequired(), validators.length(max=25)])
+        person_a_last_name = SteuerlotseNameStringField(
+            label=_l('form.lotse.field_person_last_name'),
+            render_kw={'data_label': _l('form.lotse.field_person_last_name.data_label'),
+                       'max_characters': 25},
+            validators=[InputRequired(), validators.length(max=25)])
+        person_a_street = SteuerlotseStringField(
+            label=_l('form.lotse.field_person_street'),
+            render_kw={'data_label': _l('form.lotse.field_person_street.data_label'),
+                       'max_characters': 25},
+            validators=[InputRequired(), validators.length(max=25)])
+        person_a_street_number = SteuerlotseHouseNumberIntegerField(
+            label=_l('form.lotse.field_person_street_number'),
+            render_kw={'data_label': _l('form.lotse.field_person_street_number.data_label'),
+                       'max_characters': 4},
+            validators=[InputRequired(), IntegerLength(max=4)])
+        person_a_street_number_ext = SteuerlotseStringField(
+            label=_l('form.lotse.field_person_street_number_ext'),
+            render_kw={'data_label': _l('form.lotse.field_person_street_number_ext.data_label'),
+                       'max_characters': 6},
+            validators=[validators.length(max=6)])
+        person_a_address_ext = SteuerlotseStringField(
+            label=_l('form.lotse.field_person_address_ext'),
+            render_kw={'data_label': _l('form.lotse.field_person_address_ext.data_label'),
+                       'max_characters': 25},
+            validators=[validators.length(max=25)])
+        person_a_plz = SteuerlotseNumericStringField(
+            label=_l('form.lotse.field_person_plz'),
+            render_kw={'data_label': _l('form.lotse.field_person_plz.data_label'),
+                       'max_characters': 5},
+            validators=[InputRequired(), DecimalOnly(), validators.length(max=5)])
+        person_a_town = SteuerlotseStringField(
+            label=_l('form.lotse.field_person_town'),
+            render_kw={'data_label': _l('form.lotse.field_person_town.data_label'),
+                       'max_characters': 25},
+            validators=[InputRequired(), validators.length(max=20)])
```

Comment:
Maximum character length is different here (25 != 20). Why does the render_kw also need the info? Isn't the validator enough?

Reply:
Ah a very good point you bring up. I hope this will be cleaner once we have react. So the thing is: The validator makes sure that there are only 20 chars submitted. The `max_characters` in the `render_kw` however, only sets the visual width of the field. (we have some fields that appear smaller, e.g. "Hausnummer" to indicate how many chars are required. The town apparently wasn't supposed to look as small as 20 chars but as 25 chars. 🤷‍♀️